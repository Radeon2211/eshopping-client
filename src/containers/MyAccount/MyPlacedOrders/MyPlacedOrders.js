import { useEffect } from 'react';
import { useSelector } from 'react-redux';
import * as SC from '../MyAccount.sc';
import Orders from '../../../components/Orders/Orders';
import Heading from '../../../components/UI/Heading/Heading';
import { scrollToTop } from '../../../shared/utility/utility';
import MetaDescriptor from '../../../components/MetaDescriptor/MetaDescriptor';
import useLastLocation from '../../../shared/hooks/useLastLocation';
import { OrderType } from '../../../shared/types/types';

export default function MyPlacedOrders() {
  const lastLocation = useLastLocation();

  const placedOrders = useSelector((state) => state.auth.placedOrders);

  useEffect(() => {
    if (!lastLocation?.pathname.startsWith('/order/')) {
      scrollToTop();
    }
  }, [lastLocation]);

  return (
    <SC.OrdersWrapper>
      <MetaDescriptor
        title="Your placed orders - E-Shopping"
        description="Check out orders that you placed"
      />
      <Heading $variant="h3">My placed orders</Heading>
      <Orders orders={placedOrders} type={OrderType.PLACED_ORDERS} />
    </SC.OrdersWrapper>
  );
}
