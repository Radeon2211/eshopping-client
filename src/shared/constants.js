import * as Yup from 'yup';
import { getCountries } from 'country-fns';

export const anyStringRule = Yup.string().min(1).required();

export const userRules = {
  email: Yup.string().email().max(320).trim().required(),
  hideEmail: Yup.bool(),
  username: Yup.string().min(3).max(20).trim().required(),
  password: Yup.string().min(7).max(64).trim().required(),
  firstName: Yup.string().max(60).trim().required(),
  lastName: Yup.string().max(80).trim().required(),
  phonePrefix: Yup.object()
    .shape({
      value: Yup.string().required(),
      label: Yup.string().required(),
    })
    .nullable()
    .required(),
  phoneNumber: Yup.string()
    .min(5)
    .max(15)
    .matches(/^[\d-]{5,15}$/)
    .trim()
    .required(),
  hidePhone: Yup.bool(),
  street: Yup.string().max(60).trim().required(),
  zipCode: Yup.string().max(12).trim().required(),
  city: Yup.string().max(100).trim().required(),
  country: Yup.object()
    .shape({
      value: Yup.string().required(),
      label: Yup.string().required(),
    })
    .nullable()
    .required(),
};

export const productConditions = {
  NEW: 'new',
  USED: 'used',
  NOT_APPLICABLE: 'not_applicable',
};

export const productRules = {
  name: Yup.string().max(150).trim().required(),
  price: Yup.number().min(0.01).max(1000000).required(),
  quantity: Yup.number().min(1).max(100000).required(),
  condition: Yup.string()
    .oneOf([productConditions.NEW, productConditions.USED, productConditions.NOT_APPLICABLE])
    .required(),
  description: Yup.string().max(600).trim(),
};

export const listOfAreaCodes = getCountries().map(({ name, dial }) => {
  const finalValue = `+${dial} ${name.split('(')[0].trim()}`;
  return {
    value: dial,
    label: finalValue,
  };
});

export const listOfCountries = getCountries().map(({ name }) => {
  const finalValue = name.split('(')[0].trim();
  return {
    value: finalValue,
    label: finalValue,
  };
});

export const productPages = {
  ALL_PRODUCTS: 'ALL_PRODUCTS',
  MY_PRODUCTS: 'MY_PRODUCTS',
  USER_PRODUCTS: 'USER_PRODUCTS',
};

export const itemTypes = {
  CART: 'CART',
  ORDER: 'ORDER',
  TRANSACTION: 'TRANSACTION',
};

export const listItemTypes = {
  PRODUCT: 'PRODUCT',
  ORDER: 'ORDER',
};

export const orderTypes = {
  PLACED_ORDERS: 'PLACED_ORDERS',
  SELL_HISTORY: 'SELL_HISTORY',
};

export const inputKinds = {
  INPUT: 'INPUT',
  TEXTAREA: 'TEXTAREA',
  SELECT: 'SELECT',
  RADIO: 'RADIO',
};

export const defaultProductsPerPage = 10;

export const defaultOrdersPerPage = 6;

export const defaultAppPath = '/products?p=1';

export const filtersActions = {
  INIT_STATE: 'INIT_STATE',
  SET_SORT_BY: 'SET_SORT_BY',
  SET_MIN_PRICE: 'SET_MIN_PRICE',
  SET_MAX_PRICE: 'SET_MAX_PRICE',
  SET_CONDITION: 'SET_CONDITION',
};

export const sliderPositionsActions = {
  SET_BOTH: 'SET_BOTH',
  SET_LEFT: 'SET_LEFT',
  SET_RIGHT: 'SET_RIGHT',
};

export const modalTypes = {
  ABOUT_WEBSITE: 'ABOUT_WEBSITE',
  ADD_ADMIN: 'ADD_ADMIN',
  ADD_PRODUCT: 'ADD_PRODUCT',
  BUY_PRODUCTS: 'BUY_PRODUCTS',
  CART_ITEM_ADDED: 'CART_ITEM_ADDED',
  CHANGE_NAME: 'CHANGE_NAME',
  CHANGE_EMAIL: 'CHANGE_EMAIL',
  CHANGE_USERNAME: 'CHANGE_USERNAME',
  CHANGE_PHONE_NUMBER: 'CHANGE_PHONE_NUMBER',
  CHANGE_ADDRESS: 'CHANGE_ADDRESS',
  CHANGE_PASSWORD: 'CHANGE_PASSWORD',
  CHANGE_CONTACTS: 'CHANGE_CONTACTS',
  CHANGE_DELIVERY_ADDRESS: 'CHANGE_DELIVERY_ADDRESS',
  CLEAR_CART: 'CLEAR_CART',
  DELETE_ACCOUNT: 'DELETE_ACCOUNT',
  DELETE_PRODUCT: 'DELETE_PRODUCT',
  EDIT_PRODUCT: 'EDIT_PRODUCT',
  LOGIN: 'LOGIN',
  REMOVE_ADMIN: 'REMOVE_ADMIN',
  RESET_PASSWORD: 'RESET_PASSWORD',
  PENDING_USER_INFO: 'PENDING_USER_INFO',
  SEND_VERIFICATION_LINK: 'SEND_VERIFICATION_LINK',
  SIGNUP: 'SIGNUP',
};

export const sortProductsOptions = [
  { value: undefined, label: 'Default sorting' },
  { value: 'price:asc', label: `Price - ascending` },
  { value: 'price:desc', label: 'Price - descending' },
  { value: 'name:asc', label: 'Name - A to Z' },
  { value: 'name:desc', label: 'Name - Z to A' },
];

export const sortOrdersOptions = [
  { value: 'createdAt:desc', label: 'Date - from newest' },
  { value: 'createdAt:asc', label: 'Date - from oldest' },
  { value: 'overallPrice:asc', label: `Price - ascending` },
  { value: 'overallPrice:desc', label: 'Price - descending' },
];

export const singleInfoNames = {
  USERNAME: 'Username',
  NAME: 'Name',
  EMAIL: 'Email',
  PHONE_NUMBER: 'Phone number',
  ADDRESS: 'Address',
  CONTACTS: 'Contacts',
};

export const updateCartActions = {
  INCREMENT: 'INCREMENT',
  DECREMENT: 'DECREMENT',
  NUMBER: 'NUMBER',
};

export const defaultErrorMessage = 'Something went wrong';

export const productsPerPageControllerOptions = [
  { value: 1, label: '1' },
  { value: 2, label: '2' },
  { value: 3, label: '3' },
  { value: 4, label: '4' },
  { value: 5, label: '5' },
  { value: 10, label: '10' },
  { value: 15, label: '15' },
  { value: 20, label: '20' },
  { value: 25, label: '25' },
];

export const defaultScrollToConfig = {
  top: 0,
  left: 0,
  behavior: 'smooth',
};

export const userStatuses = {
  ACTIVE: 'active',
  PENDING: 'pending',
};
