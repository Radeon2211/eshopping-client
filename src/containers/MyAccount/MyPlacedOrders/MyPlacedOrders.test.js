import { waitFor, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import MyPlacedOrders from './MyPlacedOrders';
import {
  createOrder,
  createTransactionAndOrderProdItem,
  renderAppPart,
} from '../../../shared/testUtility/testUtility';
import { defaultScrollToConfig } from '../../../shared/constants';
import useLastLocation from '../../../shared/hooks/useLastLocation';

const mockStore = configureMockStore([thunk]);

const products = [
  createTransactionAndOrderProdItem({
    _id: 'p1',
    price: 80.2,
    quantity: 4,
    name: 'product1',
    photo: true,
    seller: {
      username: 'sellerUser',
    },
  }),
];

const defaultStore = mockStore({
  auth: {
    placedOrders: [
      createOrder({
        _id: 'o1',
        products,
        seller: {
          username: 'sellerUser',
        },
        buyer: {
          username: 'buyerUser',
        },
        overallPrice: 320.8,
      }),
    ],
    orderCount: 1,
  },
  ui: {
    isDataLoading: false,
  },
});

const setUp = () => {
  return renderAppPart(<MyPlacedOrders />, {
    pathname: '/my-account/placed-orders',
    search: '?p=1',
    store: defaultStore,
  });
};

jest.mock('../../../shared/hooks/useLastLocation', () => ({
  __esModule: true,
  default: jest.fn(),
}));

beforeAll(() => {
  window.scrollTo = jest.fn();
});

describe('<MyPlacedOrders />', () => {
  describe('check how renders', () => {
    it('should render everything correctly with given default data', async () => {
      setUp();
      expect(screen.getAllByTestId('OrderList-single-order')).toHaveLength(1);
      await waitFor(() => {
        expect(document.title).toEqual('Your placed orders - E-Shopping');
      });
    });
  });

  describe('check useEffect()', () => {
    it('should call scrollTo() if last rendered route was OrderDetails', () => {
      useLastLocation.mockImplementation(() => ({
        pathname: '/order/o1',
      }));
      setUp();
      expect(window.scrollTo).not.toHaveBeenCalled();
    });

    it('should NOT call scrollTo() if last rendered route was NOT OrderDetails', () => {
      useLastLocation.mockImplementation(() => ({
        pathname: '/cart',
      }));
      setUp();
      expect(window.scrollTo).toHaveBeenCalledWith(defaultScrollToConfig);
    });
  });
});
