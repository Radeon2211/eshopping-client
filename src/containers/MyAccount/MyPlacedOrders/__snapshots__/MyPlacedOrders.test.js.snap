// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MyPlacedOrders /> check how renders should render everything correctly with given default data 1`] = `
<DocumentFragment>
  .c0 {
  max-width: 100%;
  margin: 0 auto;
  width: 80rem;
}

.c2 {
  background-color: #FAFAFA;
  padding: 2.4rem;
  width: 100%;
}

.c5 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c5 .arrow {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-align-self: stretch;
  -ms-flex-item-align: stretch;
  align-self: stretch;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin: 0 0.8rem;
}

.c5 .hide-arrow {
  opacity: 0;
  pointer-events: none;
}

.c5 .form-number {
  margin: 0 0.8rem;
}

.c6 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c6 > svg {
  -webkit-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
  -webkit-transition: -webkit-transform 0.2s;
  -webkit-transition: transform 0.2s;
  transition: transform 0.2s;
  fill: #777;
  width: 1.7rem;
  height: 1.7rem;
  -webkit-transform: rotate(180deg);
  -ms-transform: rotate(180deg);
  transform: rotate(180deg);
}

.c9 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c9 > svg {
  -webkit-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
  -webkit-transition: -webkit-transform 0.2s;
  -webkit-transition: transform 0.2s;
  transition: transform 0.2s;
  fill: #777;
  width: 1.7rem;
  height: 1.7rem;
}

.c8 {
  display: inline-block;
  font-style: normal;
  font-weight: normal;
  -webkit-letter-spacing: 0px;
  -moz-letter-spacing: 0px;
  -ms-letter-spacing: 0px;
  letter-spacing: 0px;
  margin-bottom: 0;
  margin-left: 0.8rem;
  margin-right: 0.8rem;
  overflow: visible;
  text-overflow: clip;
  white-space: normal;
  font-size: 1.5rem;
}

.c13 {
  display: inline-block;
  font-style: normal;
  font-weight: normal;
  -webkit-letter-spacing: 0px;
  -moz-letter-spacing: 0px;
  -ms-letter-spacing: 0px;
  letter-spacing: 0px;
  margin-bottom: 0;
  margin-left: 0;
  margin-right: 0;
  overflow: visible;
  text-overflow: clip;
  white-space: normal;
  font-size: 1.5rem;
}

.c15 {
  display: inline-block;
  font-style: normal;
  font-weight: normal;
  -webkit-letter-spacing: 0px;
  -moz-letter-spacing: 0px;
  -ms-letter-spacing: 0px;
  letter-spacing: 0px;
  margin-bottom: 0;
  margin-left: 0;
  margin-right: 0;
  overflow: visible;
  text-overflow: clip;
  white-space: normal;
  font-size: 1.4rem;
}

.c23 {
  display: inline-block;
  font-style: normal;
  font-weight: normal;
  -webkit-letter-spacing: 0px;
  -moz-letter-spacing: 0px;
  -ms-letter-spacing: 0px;
  letter-spacing: 0px;
  margin-bottom: 0;
  margin-left: 0;
  margin-right: 0;
  overflow: visible;
  text-overflow: clip;
  white-space: normal;
  color: #777;
  font-size: 1.4rem;
}

.c24 {
  display: inline-block;
  font-style: normal;
  font-weight: normal;
  -webkit-letter-spacing: 0px;
  -moz-letter-spacing: 0px;
  -ms-letter-spacing: 0px;
  letter-spacing: 0px;
  margin-bottom: 0;
  margin-left: 0;
  margin-right: 0;
  overflow: visible;
  text-overflow: clip;
  white-space: normal;
  font-size: 1.7rem;
}

.c27 {
  display: inline-block;
  font-style: normal;
  font-weight: normal;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  margin-bottom: 0;
  margin-left: 0;
  margin-right: 0.8rem;
  overflow: visible;
  text-overflow: clip;
  white-space: normal;
  color: #777;
  font-size: 1.5rem;
}

.c28 {
  display: inline-block;
  font-style: normal;
  font-weight: normal;
  -webkit-letter-spacing: 0px;
  -moz-letter-spacing: 0px;
  -ms-letter-spacing: 0px;
  letter-spacing: 0px;
  margin-bottom: 0;
  margin-left: 0;
  margin-right: 0;
  overflow: visible;
  text-overflow: clip;
  white-space: normal;
  font-size: 2.2rem;
}

.c30 {
  display: inline-block;
  font-style: normal;
  font-weight: normal;
  -webkit-letter-spacing: 0px;
  -moz-letter-spacing: 0px;
  -ms-letter-spacing: 0px;
  letter-spacing: 0px;
  margin-bottom: 0;
  margin-left: 0;
  margin-right: 0;
  overflow: visible;
  text-overflow: clip;
  white-space: normal;
  color: #777;
  font-size: 1.3rem;
}

.c7 {
  background-color: #FAFAFA;
  border: 1px solid #777;
  border-radius: 1px;
  box-sizing: content-box;
  font-size: 1.5rem;
  outline: none;
  padding: 0.8rem 0;
  text-align: center;
  width: 5rem;
}

.c7::-webkit-outer-spin-button,
.c7::-webkit-inner-spin-button {
  margin: 0;
  -webkit-appearance: none;
}

.c31 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 1.5rem;
}

.c31 .arrow {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-align-self: stretch;
  -ms-flex-item-align: stretch;
  align-self: stretch;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin: 0 0.8rem;
}

.c31 .hide-arrow {
  opacity: 0;
  pointer-events: none;
}

.c31 .ellipsis {
  margin: 0 0.8rem;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
}

.c31 .of {
  margin: 0 0.8rem;
}

.c31 .number-link {
  padding: 0.8rem;
  -webkit-transition: color 0.1s;
  transition: color 0.1s;
}

.c31 .number-link:hover {
  color: #06AD96;
}

.c31 .number-link.active {
  color: #22A7F0;
  font-weight: 700;
}

.c31 .number-link.active:hover {
  color: #22A7F0;
  cursor: default;
}

.c29 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin-top: 2.4rem;
  padding: 0 1.2rem 0 0;
}

.c10 {
  position: relative;
}

.c11 {
  padding: 2.4rem 0;
  border-bottom: 1px solid #BFBFBF;
}

.c11:first-child {
  padding-top: 0;
}

.c14 {
  color: #06AD96 !important;
  -webkit-transition: color 0.1s;
  transition: color 0.1s;
}

.c14:hover {
  color: #009680 !important;
  cursor: pointer;
}

.c3 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin-bottom: 2.4rem;
  padding: 0 1.2rem 0 0;
}

.c12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin-top: 0;
  margin-right: 0;
  margin-bottom: 0;
  margin-left: 0;
  width: 100%;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
}

.c12 > *:not(:last-child) {
  margin-right: 2.4rem;
}

.c16 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  margin-top: 0;
  margin-right: 0;
  margin-bottom: 0;
  margin-left: 0;
  width: 100%;
}

.c16 > *:not(:last-child) {
  margin-right: 1.2rem;
}

.c18 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-top: 0;
  margin-right: 0;
  margin-bottom: 0;
  margin-left: 0;
  width: 5;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c18 > *:not(:last-child) {
  margin-right: 0;
}

.c21 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin-top: 0;
  margin-right: 0;
  margin-bottom: 0;
  margin-left: 0;
  width: 100%;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c21 > *:not(:last-child) {
  margin-right: 0;
}

.c25 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin-top: 2.4rem;
  margin-right: 0;
  margin-bottom: 0;
  margin-left: 0;
  width: 100%;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c25 > *:not(:last-child) {
  margin-right: 2.4rem;
}

.c17 {
  padding-top: 2.4rem;
}

.c17:last-child {
  padding-bottom: 0;
}

.c20 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c20 > *:not(:last-child) {
  margin-right: 0.8rem;
}

.c20 .name {
  -webkit-flex: 2;
  -ms-flex: 2;
  flex: 2;
  font-size: 1.5rem;
}

.c20 .name > a {
  -webkit-transition: color 0.1s;
  transition: color 0.1s;
}

.c20 .name > a:hover {
  color: #06AD96;
}

.c19 {
  max-height: 5rem;
  min-width: 5rem;
  width: 5rem;
}

.c19 img {
  max-height: 5rem;
  max-width: 5rem;
  object-fit: cover;
}

.c26 {
  box-shadow: 0 0 3px rgba(0,0,0,0.03);
  border-radius: 1px;
  cursor: pointer;
  font-weight: 700;
  line-height: 1.2;
  -webkit-letter-spacing: 2px;
  -moz-letter-spacing: 2px;
  -ms-letter-spacing: 2px;
  letter-spacing: 2px;
  outline: none;
  position: relative;
  text-align: center;
  text-transform: uppercase;
  -webkit-transition: all 0.1s;
  transition: all 0.1s;
  font-size: 1.4rem;
  padding: 0.8rem 1.2rem;
  background-color: transparent;
  border: 2px solid #1E8BC3;
  color: #1E8BC3;
}

.c26:hover {
  border-color: #22A7F0;
  color: #22A7F0;
}

.c4 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  margin-right: 1.2rem;
}

.c4 .label {
  font-size: 1.5rem;
  margin-right: 1.2rem;
}

.c4 .select {
  font-size: 1.4rem;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  max-width: 22rem;
  z-index: 100;
}

.c1 {
  font-size: 2.4rem;
  font-family: Noto Sans,sans-serif;
  line-height: 1;
  margin-bottom: 2.4rem;
  margin-top: 0;
  text-align: left;
}

@media only screen and (max-width:56.25em) {
  .c2 {
    padding: 2.4rem 1.2rem;
  }
}

@media only screen and (max-width:37.5em) {
  .c29 {
    padding: 0;
  }
}

@media only screen and (max-width:37.5em) {
  .c3 {
    padding: 0;
  }
}

@media only screen and (max-width:56.25em) {
  .c20 {
    -webkit-align-items: start;
    -webkit-box-align: start;
    -ms-flex-align: start;
    align-items: start;
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
  }

  .c20 > *:not(:last-child) {
    margin-right: 0;
    margin-bottom: 0.8rem;
  }

  .c20 .name {
    -webkit-flex: initial;
    -ms-flex: initial;
    flex: initial;
  }
}

@media only screen and (max-width:56.25em) {
  .c22 {
    -webkit-flex: initial;
    -ms-flex: initial;
    flex: initial;
  }
}

<div
    class="c0"
  >
    <h3
      class="c1"
      font-size=""
    >
      My placed orders
    </h3>
    <section
      class="c2"
    >
      <div
        class="c3"
      >
        <div
          class="c4"
        >
          <label
            class="label"
            for="sortOption"
          >
            Sort
          </label>
          <div
            class="select css-2b097c-container"
            id="sortOption"
          >
            <span
              aria-atomic="false"
              aria-live="polite"
              aria-relevant="additions text"
              class="css-1f43avz-a11yText-A11yText"
            />
            <div
              class=" css-l640om-control"
            >
              <div
                class=" css-g1d714-ValueContainer"
              >
                <div
                  class=" css-1uccc91-singleValue"
                >
                  Date - from newest
                </div>
                <input
                  aria-autocomplete="list"
                  class="css-wmatm6-dummyInput-DummyInput"
                  id="react-select-2-input"
                  readonly=""
                  tabindex="0"
                  value=""
                />
              </div>
              <div
                class=" css-1hb7zxy-IndicatorsContainer"
              >
                <span
                  class=" css-1okebmr-indicatorSeparator"
                />
                <div
                  aria-hidden="true"
                  class=" css-ml9fjp-indicatorContainer"
                >
                  <svg
                    aria-hidden="true"
                    class="css-tj5bde-Svg"
                    focusable="false"
                    height="20"
                    viewBox="0 0 20 20"
                    width="20"
                  >
                    <path
                      d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                    />
                  </svg>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="c5"
          data-testid="InputPagination"
        >
          <a
            class="arrow hide-arrow"
            data-testid="InputPagination-left-arrow"
          >
            <div
              class="c6"
              color=""
            >
              <svg>
                arrow.svg
              </svg>
            </div>
          </a>
          <form
            class="form-number"
          >
            <input
              class="c7"
              data-testid="NumberInput-page"
              name="page"
              type="number"
              value="1"
            />
          </form>
          <span
            class="c8"
            color=""
            display="inline-block"
            overflow="visible"
            size="3"
            spacing="0px"
            transform=""
          >
            of
          </span>
          <span
            class="c8"
            color=""
            data-testid="InputPagination-number-of-pages"
            display="inline-block"
            overflow="visible"
            size="3"
            spacing="0px"
            transform=""
          >
            1
          </span>
          <a
            class="arrow hide-arrow"
            data-testid="InputPagination-right-arrow"
          >
            <div
              class="c9"
              color=""
            >
              <svg>
                arrow.svg
              </svg>
            </div>
          </a>
        </div>
      </div>
      <div
        class="c10"
      >
        <div
          class="c11"
        >
          <div
            class="c12"
            direction="row"
            spacing="3"
            width="100%"
            wrap="nowrap"
          >
            <span
              class="c13"
              color=""
              display="inline-block"
              overflow="visible"
              size="3"
              spacing="0px"
              transform=""
            >
              <span>
                seller 
              </span>
              <a
                data-testid="OrderList-user-link"
              >
                <span
                  class="c14"
                >
                  sellerUser
                </span>
              </a>
            </span>
            <span
              class="c15"
              color=""
              display="inline-block"
              overflow="visible"
              size="2"
              spacing="0px"
              transform=""
            >
              28 Feb 2021, 22:13
            </span>
          </div>
          <div
            class="c16 c17"
            data-testid="TransactionAndOrderProdItem"
            direction="row"
            spacing="2"
            width="100%"
            wrap="nowrap"
          >
            <a
              data-testid="TransactionAndOrderProdItem-product-link-photo"
            >
              <div
                class="c18 c19"
                data-testid="ProductThumbnail"
                direction="row"
                height="5"
                spacing=""
                width="5"
                wrap="nowrap"
              >
                <img
                  alt="product1"
                  data-testid="ProductThumbnail-img"
                  src="http://192.168.1.109:4000/orders/o1/p1/photo"
                />
              </div>
            </a>
            <div
              class="c20"
            >
              <span
                class="name"
              >
                <a
                  data-testid="TransactionAndOrderProdItem-product-link-name"
                >
                  product1
                </a>
              </span>
              <div
                class="c21 c22"
                direction="row"
                spacing=""
                width="100%"
                wrap="nowrap"
              >
                <span
                  class="c23"
                  color="#777"
                  data-testid="TransactionAndOrderProdItem-product-price-per-piece"
                  display="inline-block"
                  overflow="visible"
                  size="2"
                  spacing="0px"
                  transform=""
                >
                  4 x $80.20
                </span>
                <span
                  class="c24"
                  color=""
                  display="inline-block"
                  overflow="visible"
                  size="4"
                  spacing="0px"
                  transform=""
                >
                  $320.80
                </span>
              </div>
            </div>
          </div>
          <div
            class="c25"
            direction="row"
            spacing="3"
            width="100%"
            wrap="nowrap"
          >
            <a
              data-testid="OrderList-order-details-link"
            >
              <button
                class="c26"
                color="blue"
                type="button"
              >
                details
              </button>
            </a>
            <div>
              <span
                class="c27"
                color="#777"
                display="inline-block"
                overflow="visible"
                size="3"
                spacing="1px"
                transform=""
              >
                TOTAL
              </span>
              <span
                class="c28"
                color=""
                display="inline-block"
                overflow="visible"
                size="5"
                spacing="0px"
                transform=""
              >
                $320.80
              </span>
            </div>
          </div>
        </div>
      </div>
      <div
        class="c29"
      >
        <span
          class="c30"
          color="#777"
          data-testid="PaginationCounter"
          display="inline-block"
          overflow="visible"
          size="1"
          spacing="0px"
          transform=""
        >
          1 - 1 of 1 order
        </span>
        <div
          class="c31"
          data-testid="NumberPagination"
        >
          <a
            class="arrow hide-arrow"
            data-testid="NumberPagination-left-arrow"
          >
            <div
              class="c6"
              color=""
            >
              <svg>
                arrow.svg
              </svg>
            </div>
          </a>
          <a
            class="number-link active"
            data-testid="NumberPagination-page1"
          >
            1
          </a>
          <span
            class="of"
          >
            of
          </span>
          <span
            class="of"
            data-testid="NumberPagination-number-of-pages"
          >
            1
          </span>
          <a
            class="arrow hide-arrow"
            data-testid="NumberPagination-right-arrow"
          >
            <div
              class="c9"
              color=""
            >
              <svg>
                arrow.svg
              </svg>
            </div>
          </a>
        </div>
      </div>
    </section>
  </div>
</DocumentFragment>
`;
